### PR TITLE
Replace CaseFactory with CaseHelper

### PR DESCRIPTION
**Pull request by kaapstorm**
## Technical Summary

Uses the new `CaseHelper` class to simplify operations on cases, and to move away from `CaseFactory`.

**NOTE:** At the time this PR is opened, the base branch is not `master`. This change follows on from PR #32825 in order for migrations to be merged in sequence.

## Feature Flag

Attendance Tracking

### Safety story

Tested locally

### Automated test coverage

Covered by tests

### QA Plan

Not specifically, although the next round of Attendance Tracking QA will implicitly cover this change.

### Migrations

Small migration changes the default value of a field.

- [x] The migrations in this code can be safely applied first independently of the code

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/events/models.py [hunk 1]
```diff
@@ -272,38 +259,37 @@ def mark_attendance(self, attendee_cases, attended_datetime):
...
-                    }
+            attended_date_ext_case = self._get_ext_case_data(
+                case_type=ATTENDEE_DATE_CASE_TYPE,
+                properties={ATTENDED_DATE_CASE_PROPERTY: attended_datetime},
+                attendee_case_id=case_id
+            )
+            helper = CaseHelper(domain=self.domain)
+            helper.create_case(attended_date_ext_case)
+
+    def _get_ext_case_data(self, case_type, properties, attendee_case_id):
```
[the remark this answers is not in the document]
Comment (kaapstorm) [reply corehq/apps/events/models.py:274]: Sound like I need to rename that method! :joy:

Yes, "extension".

**Review by SmittieC — APPROVED** [review]
I would suggest renaming the `ext` to `extension`, but otherwise looks good to me

File: corehq/apps/events/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaapstorm) [reply corehq/apps/events/models.py]: a531c8986b3

[the remark this answers is not in the document]
Comment (SmittieC) [reply corehq/apps/events/models.py]: Thanks!

**Pull request merged**